From 8d62ab58c6c98688802c77819ca3dfef3682b707 Mon Sep 17 00:00:00 2001
From: Tomas Zak <tomas.zak@turris.com>
Date: Thu, 29 Aug 2024 19:56:53 +0200
Subject: [PATCH] mwlwifi: fix compilation warning with 64-bit system

This patch fixing compilation warning on 64-bit system during debugfs compilation
More info:
https://github.com/kaloz/mwlwifi/pull/413/commits/44419b1feae3eedda21cdc71da9acb611ca1a6fd
---
 .../008-fix-debugfs-compilation-warning.patch     | 15 +++++++++++++++
 1 file changed, 15 insertions(+)
 create mode 100644 package/kernel/mwlwifi/patches/008-fix-debugfs-compilation-warning.patch

diff --git a/package/kernel/mwlwifi/patches/008-fix-debugfs-compilation-warning.patch b/package/kernel/mwlwifi/patches/008-fix-debugfs-compilation-warning.patch
new file mode 100644
index 0000000000..81ef030b87
--- /dev/null
+++ b/package/kernel/mwlwifi/patches/008-fix-debugfs-compilation-warning.patch
@@ -0,0 +1,15 @@
+--- a/debugfs.c
++++ b/debugfs.c
+@@ -364,10 +364,10 @@ static ssize_t mwl_debugfs_info_read(str
+ 			 "-----------------------=>  address| address|qlen|fw_desc_cnt\n");
+ 		spin_lock_irqsave(&pcie_priv->tx_desc_lock, flags);
+ 		len += scnprintf(p + len, size - len,
+-				"wcb_base0   : %x => %8x|%8p|%4d|%d\n", get_hw_spec->wcb_base0, *((unsigned int *)le32_to_cpu(get_hw_spec->wcb_base0)),(void *)*((unsigned int *)le32_to_cpu(get_hw_spec->wcb_base0)),skb_queue_len(&pcie_priv->txq[0]),pcie_priv->fw_desc_cnt[0]);
++				"wcb_base0   : %x => %8lx|%8p|%4d|%d\n", get_hw_spec->wcb_base0, *((uintptr_t *)(uintptr_t)le32_to_cpu(get_hw_spec->wcb_base0)),(void *)*((uintptr_t *)(uintptr_t)le32_to_cpu(get_hw_spec->wcb_base0)),skb_queue_len(&pcie_priv->txq[0]),pcie_priv->fw_desc_cnt[0]);
+ 		for(i = 0; i < SYSADPT_TOTAL_TX_QUEUES - 1; i++)
+ 			len += scnprintf(p + len, size - len,
+-				"wcb_base[%2d]: %x => %8x|%8p|%4d|%d\n", i, get_hw_spec->wcb_base[i], *((unsigned int *)le32_to_cpu(get_hw_spec->wcb_base[i])),(void *)*((unsigned int *)le32_to_cpu(get_hw_spec->wcb_base[i])),skb_queue_len(&pcie_priv->txq[i + 1]),pcie_priv->fw_desc_cnt[i + 1]);
++				"wcb_base[%2d]: %x => %8lx|%8p|%4d|%d\n", i, get_hw_spec->wcb_base[i], *((uintptr_t *)(uintptr_t)le32_to_cpu(get_hw_spec->wcb_base[i])),(void *)*((uintptr_t *)(uintptr_t)le32_to_cpu(get_hw_spec->wcb_base[i])),skb_queue_len(&pcie_priv->txq[i + 1]),pcie_priv->fw_desc_cnt[i + 1]);
+ 		spin_unlock_irqrestore(&pcie_priv->tx_desc_lock, flags);
+ 	}
+ 
-- 
2.46.0

